src/server.js

```diff
@@ -1,37 +1,65 @@
 import path from 'path'
 import express from 'express'
 import React from 'react'
+import { createStore, applyMiddleware } from 'redux'
+import { Provider } from 'react-redux'
+import reduxPromiseMiddleware from 'redux-promise'
 import App from './components/App'
 import { ServerStyleSheet } from 'styled-components'
-import { renderToString, renderToStaticMarkup } from 'react-dom/server'
+import { renderToString } from 'react-dom/server'
+import { renderRoutes, matchRoutes } from 'react-router-config'
 import { env, port, ip, basename } from './config'
 import Html from './components/Html'
 import { StaticRouter } from 'react-router-dom'
+import routes from './routes'
+import reducers from './reducers/index'
+import serialize from 'serialize-javascript'
+
+const createStoreWithMiddleware = applyMiddleware(reduxPromiseMiddleware)(
+  createStore
+)
 
 const isDev = process.env.NODE_ENV === 'development'
 
 const app = express()
 app.use('/public', express.static(path.join(__dirname, '../public')))
 
 app.get('*', (req, res) => {
+  const branch = matchRoutes(routes, req.url)
+
   if (isDev) {
     global.webpackIsomorphicTools.refresh()
   }
 
-  const content = renderToString(
-    <StaticRouter location={req.params[0] || '/'} context={{}}>
-      <App />
-    </StaticRouter>
-  )
-  const sheet = new ServerStyleSheet()
-  const styles = sheet.getStyleTags() // or sheet.getStyleElement()
+  const store = createStoreWithMiddleware(reducers)
 
   const assets = global.webpackIsomorphicTools.assets()
-  const markup = <Html {...{ styles, assets, content }} />
-  const html = renderToStaticMarkup(markup)
-  const doctype = '<!doctype html>\n'
 
-  res.send(doctype + html)
+  const promises = branch.map(({ route, match }) => {
+    let fetchData = route.component.fetchData
+    return fetchData instanceof Function
```

Comment:
It's faster to check with `typeof`. However, when do you face the case when this won't be a function?